backends/gdx-backends-gwt/src/com/badlogic/gdx/backends/gwt/GwtApplication.java

```diff
@@ -538,11 +538,28 @@ native static public void consoleLog(String message) /*-{
 		console.log( "GWT: " + message );
 	}-*/;
 	
-	private native void addEventListeners () /*-{
+	private native void addEventListeners() /*-{
 		var self = this;
-		$doc.addEventListener('visibilitychange', function (e) {
-			self.@com.badlogic.gdx.backends.gwt.GwtApplication::onVisibilityChange(Z)($doc['hidden'] !== true);
-		});
+
+		var eventName = null;
+		if ("hidden" in $doc) {
+			eventName = "visibilitychange"
+		} else if ("webkitHidden" in $doc) {
+			eventName = "webkitvisibilitychange"
+		} else if ("mozHidden" in $doc) {
+			eventName = "mozvisibilitychange"
+		} else if ("msHidden" in $doc) {
+			eventName = "msvisibilitychange"
+		}
+
+		if (eventName !== null) {
+			$doc
+					.addEventListener(
+							eventName,
+							function(e) {
+								self.@com.badlogic.gdx.backends.gwt.GwtApplication::onVisibilityChange(Z)($doc['hidden'] !== true);
+							});
+		}
```

Comment:
The lines above this comment. The formatter probably doesn't play nice with jsni so best to do it manually according to the usual libgdx formatting for java.